src/mapblock.h

```diff
@@ -433,11 +432,19 @@ class MapBlock
 	static const u32 nodecount = MAP_BLOCKSIZE * MAP_BLOCKSIZE * MAP_BLOCKSIZE;
 
 private:
+#if BUILD_UNITTESTS
+	// access to data, tryConvertToMonoBlock, deconvertMonoblock
+	friend class TestMapBlock;
+#endif
+
 	/*
 		Private methods
 	*/
 
 	void deSerialize_pre22(std::istream &is, u8 version, bool disk);
+	void tryConvertToMonoblock();
```

Comment:
Perhaps 'tryShrinkNodeArray` and `expandNodeArray` would be better names... I'm not good at naming.